java/spark/src/main/java/com/logicalclocks/hsfs/spark/engine/SparkEngine.java

```diff
@@ -1097,5 +1097,14 @@ protected String makeQueryName(String queryName, FeatureGroupBase featureGroup)
     }
     return queryName;
   }
-
+  
+  public void closeSparkSession() {
+    if (getSparkSession() != null) {
+      try {
+        getSparkSession().sparkContext().stop();
```

Comment:
Why are you calling `stop()` on the spark context and not `close()` on the Spark session? 
The spark context is a "component" of the spark session. You want to close the entire session, not a single component.

Reply:
On the Python side you are correctly closing the session and not just the context.